metabase/metabase
# Whats new notification

Epic: https://github.com/metabase/metabase/issues/33165

Closes https://github.com/metabase/metabase/issues/33230

Product doc: https://www.notion.so/metabase/Provide-a-heads-up-when-Metabase-is-upgraded-with-what-s-new-44a28b55624f48e28dc3f616ce7cda0f

This PR adds a notification that links to the release notes.
The notification is shown if:
- the current version is present in `version-info` (this excludes snapshots, release candidates and betas)
- there's a version in `version-info` that:
  - has a announcement_url (this will be added only to major/golden releases)
  - is > than the last acknowledged version
  - is <= the current version
  
The notification is dismissible per user and will not be dismissed automatically.
It should not cover any content in the nav bar.

## How to test locally with cypress:
This features requires the current version to be released (to be in version-info), when testing/developing locally this is never the case as a special version name is used.

I created a cypress test that sets up the required settings so that we can test it locally, to use it we need to:
- run metabase locally on this branch (`yarn dev`) in a terminal
- edit the file `e2e/test/scenarios/onboarding/navbar/whats-new.cy.spec.js` and replace `it.skip` with `it.only` on line 27
- run cypress (`yarn test-cypress-open`) in another terminal (this will take some time)
- search for the correct test by filtering for `what`, you should see the following:
 
<img width="1499" alt="image" src="https://github.com/metabase/metabase/assets/1914270/fa3f8e43-0c10-4fd8-a52c-3b58c393b3df">

- click on "whats-new.cy"

- cypress should run the test and show you metabase's home with the notification showing up

You should be able to dismiss the notification, and the link should link to `metabase.com/releases/48` (which is 404, that's ok)



--- 
 


<img width="370" alt="image" src="https://github.com/metabase/metabase/assets/1914270/fd2ade22-b12c-407d-a83b-88bb00a64552">
